interpreter/cling/test/Prompt/decls.C

```diff
@@ -16,6 +16,33 @@ struct U{void f() const {};} uu;
 
 struct V{V(): v(12) {}; int v; } vv;
 
+struct typeA {
+  typeA(int i): num{i} {};
+  typeA();
+  ~typeA();
+
+  typeA& operator()(int k);
+  const typeA& operator[](int k) const;
+
+  int num;
+};
+
+typeA::typeA() = default;
+typeA::~typeA() {};
+::typeA& ::typeA::operator()(int k) { return *this; }
+const typeA& typeA::operator[](int k) const { return *this; }
+const ::typeA & operator>=(const typeA &lhs, int t) { return lhs; }
+const typeA& operator<=(::typeA const &lhs, int t) { return lhs; }
+const typeA& operator<(typeA const&lhs, int t) { return lhs; }
+const ::typeA& operator>(const typeA &lhs, int t) { return lhs; }
+typeA operator "" _tA(unsigned long long int t) { return typeA{static_cast<int>(t)}; }
+
+(*new typeA(58))(99)[4].num // CHECK:58
+((((*(new typeA(-675)) < 81) > 43) >= 78) <= -5).num  // CHECK:-675
+83_tA .num  // CHECK:83
+
```

Comment:
Can also test ostream here like
```
std::ostream& operator<<(std::ostream& os, const typeA& a) {return (os << a.num);}

std::cout << 76601_tA; // CHECK:76601
```
But that requires including `<iostream>`. Or we can just test `operator<<` parsing without `ostream` but probably >95% of use cases call `<<` with `ostream`.

Reply:
Both approaches look good to me.  In particular, we try to avoid too much dependencies in tests, but if it improves test coverage, I wouldn't object against including the `<iostream>` header.